muspinsim/__main__.py

```diff
@@ -72,26 +127,46 @@ def main(use_mpi=False):
 
     is_fitting = mpi.broadcast(is_fitting)
 
+    out_path = inp_dir
+    if args.out_dir:
+        out_path = ensure_dir_path_exists(args.out_dir)
+
     if not is_fitting:
         # No fitting
         runner = ExperimentRunner(infile, {})
         runner.run()
 
         if mpi.is_root:
             # Output
-            runner.config.save_output()
+            runner.config.save_output(name=None, path=out_path)
     else:
         fitter = FittingRunner(infile)
-        fitter.run()
+        fitter.run(name=None, path=out_path)
 
         if mpi.is_root:
-            fitter.write_report()
+            rep_dname = inp_dir
+            rep_fname = "{0}_fit_report.txt".format(inp_file_name)
+            if args.fit_report_path:
+                try:
+                    ensure_dir_path_exists(os.path.dirname(args.fit_report_path))
+                    rep_fname = os.path.basename(args.fit_report_path)
+                    rep_dname = os.path.dirname(args.fit_report_path)
+
+                # in case it was just a filename given
+                # - default directory is input dir
+                except NotADirectoryError:
+                    rep_fname = args.fit_report_path
+                    rep_dname = inp_dir
```

Comment:
```suggestion
                try:            
                    # check if directory exists, if not create it
                    ensure_dir_path_exists(os.path.dirname(args.fit_report_path))
                    rep_fname = os.path.basename(args.fit_report_path)
                    rep_dname = os.path.dirname(args.fit_report_path)
                except NotADirectoryError:
                    # if fit_report_path was just a filename use inp_dir as a default
                    rep_fname = args.fit_report_path
                    rep_dname = inp_dir
```
If above change to comments accepted, make this consistent.